### ISSUE TITLE
Add back the otlpmetric{http,grpc} exporter code

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#2807]
~Blocked by the metric SDK having a working example in the new_sdk/main branch~

Removed in https://github.com/open-telemetry/opentelemetry-go/pull/2802

- [x] internal transforms (#3065)
- [x] internal configuration (#3090)
- [x] otlpmetric exporter (#3089)
- [x] otlpmetricgrpc client (#3094)
- [x] otlpmetrichttp client (#3097)
- [x] internal integration testing
  - [x] Client integration testing (#3125)
  - [x] gRPC collector (#3126)
  - [x] HTTP collector (#3155)

**Comment by MrAlias** [comment]
Looking at the previous package structure, I do not think the `otlpmetric.Exporter` or `otlpmetric.Client` types should be re-introduced. Having this exporter type that needs an additional concept of a client is not useful to the user. It only adds to the mental strain of setting up an exporter having to know that you need a client in addition to this exporter. And in reality, the end-user never interacts with this type. They just call `New` from a client implementation package (e.g. [`otlpmetricgrpc.New`](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc#New)).

Instead of bringing this confusion forward into the next release I propose these types are removed from the `otlpmetric` package, the implementation details of a client are not exposed from the `otlpmetrichttp` or `otlpmetricgrpc` packages, and instead those packages each have the following `New` functions:

```go
func New(ctx context.Context, opts ...Option) (metric.Exporter, error)
```

**Comment by MrAlias** [comment]
Now is a good time to reduce the options included in the `otlpmetricgrpc` package. We should accept a gRPC connection that the user creates and just use that. We should stop creating and trying to manage gRPC connections with the exporter. It means we would have to track non-stable gRPC features with our stable exporter and continually try to keep up with adding the same features gRPC does.

**Comment by MrAlias** [comment]
> Now is a good time to reduce the options included in the `otlpmetricgrpc` package. We should accept a gRPC connection that the user creates and just use that. We should stop creating and trying to manage gRPC connections with the exporter. Doing so would mean we would have to track non-stable gRPC features with our stable exporter and continually try to keep up with adding the same features gRPC does.

This might not be possible if we plan to continue support for environment variables. A connection will need to be setup in that case.

**Comment by MrAlias** [comment]
> Looking at the previous package structure, I do not think the `otlpmetric.Exporter` or `otlpmetric.Client` types should be re-introduced. Having this exporter type that needs an additional concept of a client is not useful to the user. It only adds to the mental strain of setting up an exporter having to know that you need a client in addition to this exporter. And in reality, the end-user never interacts with this type. They just call `New` from a client implementation package (e.g. [`otlpmetricgrpc.New`](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc#New)).
> 
> Instead of bringing this confusion forward into the next release I propose these types are removed from the `otlpmetric` package, the implementation details of a client are not exposed from the `otlpmetrichttp` or `otlpmetricgrpc` packages, and instead those packages each have the following `New` functions:
> 
> ```go
> func New(ctx context.Context, opts ...Option) (metric.Exporter, error)
> ```

Giving this some thought, the need for consistency out-weighs the improvements this alternate approach would offer. Users already using the `otlptracegrpc` exporter would be confused by the difference more than appreciative is my guess. I think moving forward with the original design makes sense here.

**Comment by MrAlias** [comment]
Closed by #3155
